app/scripts/services/paymentReportService.js

```diff
@@ -0,0 +1,95 @@
+angular
+  .module('confRegistrationWebApp')
+  .service('paymentReportService', function PaymentReportService(
+    $cacheFactory,
+    $rootScope,
+    $http,
+    $q,
+  ) {
+    const path = function(id) {
+      return 'conferences/' + (id || '') + '/payments/report';
+    };
+
+    function getReportData(url) {
+      const defer = $q.defer();
+      $rootScope.loadingMsg = 'Loading Details';
+      $http
+        .get(url)
+        .then(function(response) {
+          var report = response.data;
+          defer.resolve(report);
+        })
+        .catch(function(error) {
+          defer.reject(error);
+        })
```

Comment:
If you wanted to rethrow you could do
```js
.catch(function(error) {
  throw error;
})
```
But you aren't doing anything with the error or transforming it in any way so you can just delete the whole `catch` block and the error will propagate.

Reply:
done